Docker pull toomanyrequests

Seeing several instances of this:

```
4s
Run docker compose -f docker-compose.yaml pull --quiet
 unifi Pulling 
 mongo Pulling 
toomanyrequests: retry-after: 985.86µs, allowed: 44000/minute
Error: Process completed with exit code 1.
```

E.g., https://github.com/alexklibisz/terraform-provider-terrifi/actions/runs/22536379738/job/65284646060

--- a/.github/workflows/ci-local.yaml
+++ b/.github/workflows/ci-local.yaml
@@ -46,9 +46,30 @@ jobs:
       - name: Unit tests
         timeout-minutes: 1
         run: task test:unit
+      - name: Compute Docker image cache key
+        id: docker-image-key
+        run: echo "hash=$(grep 'image:' docker-compose.yaml | sha256sum | cut -d' ' -f1)" >> "$GITHUB_OUTPUT"
+      - name: Cache Docker images
+        uses: actions/cache@v4
+        id: docker-cache
+        with:
+          path: /tmp/docker-images
+          key: docker-images-${{ steps.docker-image-key.outputs.hash }}
+      - name: Load cached Docker images
+        if: steps.docker-cache.outputs.cache-hit == 'true'
+        timeout-minutes: 1
+        run: |
+          for img in /tmp/docker-images/*.tar; do
+            docker load -i "$img"
+          done
       - name: Pull Docker images
+        if: steps.docker-cache.outputs.cache-hit != 'true'
         timeout-minutes: 10
-        run: docker compose -f docker-compose.yaml pull --quiet
+        run: |
+          docker compose -f docker-compose.yaml pull --quiet
+          mkdir -p /tmp/docker-images
+          docker save mongo:8.2 -o /tmp/docker-images/mongo.tar
+          docker save lscr.io/linuxserver/unifi-network-application:version-10.1.85 -o /tmp/docker-images/unifi.tar
       - name: Acceptance tests (Docker)
         timeout-minutes: 10
         run: task test:acc

